src/Data/Text/Lines/Internal.hs

```diff
@@ -243,6 +244,25 @@ lines (TextLines (Text arr off len) nls) = go off (U.toList nls)
 splitAtLine :: HasCallStack => Word -> TextLines -> (TextLines, TextLines)
 splitAtLine k = splitAtPosition (Position k 0)
 
+-- | Get line with given 0-based index, O(1).
+-- The resulting Text does not contain newlines.
+-- Returns "" if the line index is out of bounds.
+--
+-- >>> :set -XOverloadedStrings
+-- >>> map (\l -> getLine l "fя𐀀\n☺bar\n\n") [0..3]
+-- ["fя𐀀","☺bar","",""]
+--
+getLine :: Word -> TextLines -> Text
+getLine line (TextLines t@(Text arr off len) nls)
+  | intToWord (U.length nls) < line = mempty
+  | otherwise =
+    let lineIdx = wordToInt line
+    in case (nls U.!? (lineIdx - 1), nls U.!? lineIdx) of
+      (Nothing, Nothing) -> t
+      (Nothing, Just endNl) -> Text arr off (endNl - off)
```

Comment:
Isn't this case impossible? It would be better to use nested `case` to eliminate it.

Reply:
Actually it's not impossible (thanks to QuickCheck tests that made me aware of this):
For example "a\n" this triggers this branch:
```
-- >>> :set -XOverloadedStrings
-- >>> getLine 0 "a\n"
-- "a"
```

Reply:
Worth writing this explanation into the code!

Reply:
Ok, added.